controllers/machine_controller_phases.go

```diff
@@ -240,14 +241,18 @@ func (r *MachineReconciler) reconcileInfrastructure(ctx context.Context, cluster
 	// Call generic external reconciler.
 	infraReconcileResult, err := r.reconcileExternal(ctx, cluster, m, &m.Spec.InfrastructureRef)
 	if err != nil {
-		if m.Status.InfrastructureReady && strings.Contains(err.Error(), "could not find") {
-			// Infra object went missing after the machine was up and running
+		return ctrl.Result{}, err
+	}
+	if infraReconcileResult.RequeueAfter > 0 {
```

Comment:
This is kind of brittle, but TBH also the previous implementation checking the error message wasn't bullet proof...